arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py

```diff
@@ -39,130 +40,144 @@
+        # if "username" in self.cfg["arangodb"]:
+        #     logger.info("user name for connection: " +
+        #                 str(self.cfg["arangodb"][self.mscp.DB_USER_NAME]))
+        #     db_user_name = self.cfg["arangodb"][self.mscp.DB_USER_NAME]
+        # else:
+        #     db_user_name = ""
+        # if "password" in self.cfg["arangodb"]:
+        #     logger.info("A specific password was requested !")
+        #     db_password = self.cfg["arangodb"][self.mscp.DB_PASSWORD]
+        # else:
+        #     db_password = ""
+
+        # if self.mscp.DB_CONN_PROTOCOL in self.cfg["arangodb"]:
+        #     db_conn_protocol = self.cfg["arangodb"][self.mscp.DB_CONN_PROTOCOL]
+        # else:
+        #     db_conn_protocol = "http"
 
         if self.mscp.DB_REPLICATION_FACTOR in self.cfg["arangodb"]:
             db_replication_factor = self.cfg["arangodb"][
-                self.mscp.DB_REPLICATION_FACTOR
-            ]
+                self.mscp.DB_REPLICATION_FACTOR]
         else:
             db_replication_factor = None
 
-        if self.mscp.DB_ROOT_USER in self.cfg["arangodb"]:
-            logger.info("A root user was specified, persisting...")
-
-        if self.mscp.DB_ROOT_USER_PASSWORD in self.cfg["arangodb"]:
-            logger.info("A root user password was specified, persisting...")
-
-        try:
-            self.create_db(
-                db_serv_host,
-                db_serv_port,
-                db_serv_name,
-                db_end_point,
-                db_dbName,
-                db_user_name,
-                db_password,
-                db_conn_protocol,
-            )
-
-            # If you could create a DB, proceed with provisioning the graph.
-            # Otherwise you had an issue creating the database.
-            if self.db is not None:
-                self.create_enterprise_ml_graph(db_replication_factor)
-
-                if persist_conn:
-                    self.config.dump_data()
-        except Exception:
-            logger.error("Error connecting to DB, trying again...")
-            time.sleep(2)
-            self.create_db(
-                db_serv_host,
-                db_serv_port,
-                db_serv_name,
-                db_end_point,
-                db_dbName,
-                db_user_name,
-                db_password,
-                db_conn_protocol,
-            )
-            self.create_enterprise_ml_graph(db_replication_factor)
-            if persist_conn:
-                self.config.dump_data()
-            logger.error("Obtained connection with retry!")
+        # if self.mscp.DB_ROOT_USER in self.cfg["arangodb"]:
+        #     logger.info("A root user was specified, persisting...")
+
+        # if self.mscp.DB_ROOT_USER_PASSWORD in self.cfg["arangodb"]:
+        #     logger.info("A root user password was specified, persisting...")
+
+        #Provision the graph
+        self.create_enterprise_ml_graph(db_replication_factor)
```

Comment:
Would it make sense to add an optional `create_graph` boolean argument to the `__init__` function to make this optional? e.g. `create_graph: bool = True` and then make this provisioning conditional?
```
if create_graph:
    self.create_enterprise_ml_graph(db_replication_factor)
```
(just thinking out loud here)

Reply:
Very good point. I've added this.